test/e2e/config/docker.yaml

```diff
@@ -318,6 +320,21 @@ providers:
       files:
       - sourcePath: "../data/shared/main/metadata.yaml"
 
+- name: helm
```

Comment:
I'm not sure about introducing a dependency to a very specific addon provider into the CAPI E2E tests.
Also, AFAIK this is not being used by any of the CAPI E2E test, so we don't really need it

Reply:
That's fair, I mostly added this because I wanted to be confident that this PR would allow CAAPH to work in e2e tests, i.e. get the controller running and CRDs installed. Though one question I have would be -- do we want to enable the Helm add-on provider by default? If so, I think it would make sense for it to be a part of the e2e test to ensure the out-of-box behavior is working.

Reply:
installing this provider by default is a breaking change that will impact all the users, so I'm -1 for now.

WRT to E2E test, currently the CI team is working hard to chase down flakes, and one that is really tricky is github rate limiting when downloading images on test infra; until that is fixed, I don't think we should download an additional image for CAAPH if we are not actually using it

Also, the add-on provider is just another provider type. once the scaffolding for flags is in place, everything else is shared code that is already tested

Reply:
I think if there is a need for e2e tests they can be implemented in the caaph provider repo. Just like infra providers do it today

Reply:
Gotcha, I suppose I can remove it in that case. I discussed this with @CecileRobertMichon and the main thing was that we wanted to be confident that the changes in this PR would work and that subsequent PRs to CAPI wouldn't break the add-on installation.